### DISCUSSION TITLE
nRF52840 and SX1262 GPIO Mapping

### DISCUSSION THREAD

**Discussion by PricelessToolkit** [meshtastic/firmware#4203] [Q&A] [answered]
Hello Meshmastik community,

I need some help. I'm trying to find the default GPIO mapping between the nRF52840 and SX1262. Is there a schematic available that shows this mapping? Additionally, I'm curious if it's possible to change this mapping in the firmware.

Thank you!

**Comment by garthvh** [comment]
You will find a variant file for each hardware model in the source identifying the pins used.

**Reply by PricelessToolkit** [reply]
Thank you very much!!

I am designing my own board based on a different module "nRF52840+SX1262"  so I wanted to make sure it will work.

It has a slightly different schematic:

- 16 - DIO1
- 15 - DIO3
- 19 - BUSY
- 12 - SX_NRESET
- 17 - MISO
- 20 - MOSI
- 21 - SCK
- 22 - NSS

Mechmastic not using NSS? 

```c

T-ECHO

/*
 * Lora radio
 */

#define USE_SX1262
#define USE_SX1268
#define SX126X_CS (0 + 24) // FIXME - we really should define LORA_CS instead
#define SX126X_DIO1 (0 + 20)
// Note DIO2 is attached internally to the module to an analog switch for TX/RX switching
#define SX1262_DIO3                                                                                                              \
    (0 + 21) // This is used as an *output* from the sx1262 and connected internally to power the tcxo, do not drive from the main
// CPU?
#define SX126X_BUSY (0 + 17)
#define SX126X_RESET (0 + 25)
// Not really an E22 but TTGO seems to be trying to clone that
#define SX126X_DIO2_AS_RF_SWITCH
#define SX126X_DIO3_TCXO_VOLTAGE 1.8
// Internally the TTGO module hooks the SX1262-DIO2 in to control the TX/RX switch (which is the default for the sx1262interface
// code)

// #define LORA_DISABLE_SENDING // Define this to disable transmission for testing (power testing etc...)

// #undef SX126X_CS

/*
 * eink display pins
 */

**Reply by markbirss** [reply]
Check out the meshtastic forum also - https://meshtastic.discourse.group/t/wip-diy-nrf82540-nicerf-868-4-2-inch-epaper/5552

**Reply by PricelessToolkit** [reply, marked as answer]
Thank you for the link! Once the prototype is ready, I will start a new topic in the forum. I hope the community will appreciate my open-source project.
